File: src/main/java/com/fengsheng/phase/MainPhaseIdle.java
package com.fengsheng.phase;

import com.fengsheng.Fsm;
import com.fengsheng.Player;
import com.fengsheng.ResolveResult;

/**
 * 出牌阶段空闲时点
 */
public record MainPhaseIdle(Player player) implements Fsm {
    @Override
    public ResolveResult resolve() {
        if (!player.isAlive()) {
            return new ResolveResult(new NextTurn(player), true);
        }
        for (Player p : player.getGame().getPlayers()) {
            p.notifyMainPhase(30);
        }
        return new ResolveResult(this, false);
    }

    @Override
    public String toString() {
        return player + "的出牌阶段";
    }
}

File: src/main/kotlin/phase/MainPhaseIdle.kt
package com.fengsheng.phase

import com.fengsheng.Config
import com.fengsheng.Player
import com.fengsheng.ProcessFsm
import com.fengsheng.ResolveResult

/**
 * 出牌阶段空闲时点
 */
data class MainPhaseIdle(
    /**
     * 表示当前回合的玩家。 **/
    override val whoseTurn: Player
) : ProcessFsm() {
    override fun resolve0(): ResolveResult? {
        if (!whoseTurn.alive) {
            return ResolveResult(NextTurn(whoseTurn), true)
        }
        for (p in whoseTurn.game!!.players) {
            p!!.notifyMainPhase(Config.WaitSecond * 4 / 3)
        }
        return null
    }

    override fun toString(): String {
        return "${whoseTurn}的出牌阶段"
    }
}